### PR TITLE
Various fixes for 2.0.0

### PR DESCRIPTION
**Pull request by Kovah**
@underdogg

Pull Request Checklist

  * [ ] My code follows the code formatting guidelines.
  * [ ] I have an issue ID for this pull request. 
  * [ ] I selected the corresponding branch.
  * [ ] I have rebased my changes on top of the corresponding branch.
  
Issue Type (Please check one or more)

  * [x] Bugfix
  * [ ] Improvement of an existing Feature 
  * [ ] New Feature

### CODE REVIEW

File: app/Modules/Clients/Controllers/ClientController.php [hunk 1]
```diff
@@ -41,6 +41,10 @@ public function index()
             ->keywords(request('search'))
             ->paginate(config('fi.resultsPerPage'));
 
+
```
Comment (Kovah) [inline app/Modules/Clients/Controllers/ClientController.php]: Pls remove unneeded lines

File: app/Events/Listeners/CheckAttachmentListener.php [hunk 2]
```diff
@@ -13,6 +13,10 @@ public function handle(CheckAttachment $event)
                 if ($attachment) {
                     $event->object->attachments()->create([
                         'user_id' => auth()->user()->id,
+                        /*
```
Comment (Kovah) [inline app/Events/Listeners/CheckAttachmentListener.php]: Comment can be removed

File: app/Modules/Clients/Views/clients/index.blade.php [hunk 3]
```diff
@@ -78,7 +78,8 @@ class="fa fa-plus"></i> @lang('ip.new')</a>
                             </thead>
 
                             <tbody>
-                            @foreach ($clients as $client)
+
+                            @forelse($clients as $client)
```
Comment (Kovah) [inline app/Modules/Clients/Views/clients/index.blade.php]: Pretty cool! Didn't know that this directive exists! 👍

File: app/Modules/Currencies/Support/CurrencyConverterFactory.php [hunk 4]
```diff
@@ -18,7 +18,15 @@ class CurrencyConverterFactory
 {
     public static function create()
     {
-        $class = 'IP\Modules\Currencies\Support\Drivers\\' . config('fi.currencyConversionDriver');
+
+$test = config('fi.currencyConversionDriver');
```
Comment (Kovah) [inline app/Modules/Currencies/Support/CurrencyConverterFactory.php]: Please properly format the code, may also rename `$test` to `$fallback_driver` which is more clear

File: app/Modules/Invoices/Controllers/InvoiceCopyController.php [hunk 5]
```diff
@@ -71,9 +72,14 @@ public function store(InvoiceStoreRequest $request)
         }
 
         // Copy the custom fields
-        $custom = collect($fromInvoice->custom)->except('invoice_id')->toArray();
-        $toInvoice->custom->update($custom);
+        /*$custom = collect($fromInvoice->custom)->except('invoice_id')->toArray();
```
Comment (Kovah) [inline app/Modules/Invoices/Controllers/InvoiceCopyController.php]: What is happening here or what was wrong?
There should be no commented code. If it's not needed anymore, remove it.

File: app/Modules/Invoices/Controllers/InvoiceEditController.php [hunk 6]
```diff
@@ -43,7 +44,8 @@ public function edit($id)
             ->with('customFields', CustomField::forTable('invoices')->get())
             ->with('returnUrl', $this->getReturnUrl())
             ->with('templates', InvoiceTemplates::lists())
-            ->with('itemCount', count($invoice->invoiceItems));
+            //items
```
Comment (Kovah) [inline app/Modules/Invoices/Controllers/InvoiceEditController.php]: Comment can be removed

File: app/Modules/Invoices/Controllers/InvoiceEditController.php [hunk 7]
```diff
@@ -34,6 +34,7 @@ class InvoiceEditController extends Controller
     public function edit($id)
     {
         $invoice = Invoice::with(['items.amount.item.invoice.currency'])->find($id);
+        $invoiceItemsCount = isset($invoice->invoiceItems) ? count($invoice->invoiceItems) : 0;
```
Comment (Kovah) [inline app/Modules/Invoices/Controllers/InvoiceEditController.php]: Please stick to a uniform variable naming. Use snake case for single vars: `$invoice_item_count`

File: app/Modules/Invoices/Controllers/InvoiceEditController.php [hunk 8]
```diff
@@ -59,7 +61,10 @@ public function update(InvoiceUpdateRequest $request, $id)
         $invoice->save();
 
         // Save the custom fields.
-        $invoice->custom->update(request('custom', []));
+        /*
```
Comment (Kovah) [inline app/Modules/Invoices/Controllers/InvoiceEditController.php]: Can be completely removed then

File: app/Modules/Layouts/Views/layouts/_header.blade.php [hunk 9]
```diff
@@ -28,9 +28,9 @@
                         <i class="fa fa-cog"></i>
                     </a>
                     <ul class="dropdown-menu">
-                        <li><a href="{{ route('addons.index') }}">@lang('ip.addons')</a></li>
+                        {{--<li><a href="{{ route('addons.index') }}">@lang('ip.addons')</a></li>--}}
```
Comment (Kovah) [inline app/Modules/Layouts/Views/layouts/_header.blade.php]: Why are they commented out?

File: database/migrations/2018_06_27_201645_v2_0_0_alpha2.php [hunk 10]
```diff
@@ -671,9 +672,9 @@ public function up()
         }
 
         // Add base payment methods
-        PaymentMethod::create(['name' => trans('fi.cash')]);
-        PaymentMethod::create(['name' => trans('fi.credit_card')]);
-        PaymentMethod::create(['name' => trans('fi.online_payment')]);
+        PaymentMethod::create(['name' => @lang('ip.cash')]);
```
Comment (Kovah) [inline database/migrations/2018_06_27_201645_v2_0_0_alpha2.php]: `@lang` is only available in blade templates, `trans()` would be correct to use here

**Comment by Kovah** [comment]
@UnderDogg Thank you for all the fixes. I added comments for some things to be corrected and so on. 👍 

A small improvement hint: create single commits for every fix. This gives you the possibility to comment on what you did without the need to "pollute" the code itself. A good example is removing code for various reasons. Instead of commenting out the code and adding a comment above it why it was removed, you just remove the code and explain why you did so in the commit message.

It's always better to submit 30 small commits instead of just one big commit.

File: app/Events/Listeners/CheckAttachmentListener.php (hunk 2 above)
Comment (UnderDogg) [reply app/Events/Listeners/CheckAttachmentListener.php]: Resolved in commit 2b00f09: added comment in commit message and removed it from the source

File: app/Modules/Currencies/Support/CurrencyConverterFactory.php (hunk 4 above)
Comment (UnderDogg) [reply app/Modules/Currencies/Support/CurrencyConverterFactory.php]: Resolved in fa16c07

File: app/Modules/Invoices/Controllers/InvoiceEditController.php (hunk 8 above)
Comment (UnderDogg) [reply app/Modules/Invoices/Controllers/InvoiceEditController.php]: Resolved in 82b2a34

File: app/Modules/Invoices/Controllers/InvoiceEditController.php (hunk 7 above)
Comment (UnderDogg) [reply app/Modules/Invoices/Controllers/InvoiceEditController.php]: Resolved in 82b2a34

File: app/Modules/Invoices/Controllers/InvoiceEditController.php (hunk 6 above)
Comment (UnderDogg) [reply app/Modules/Invoices/Controllers/InvoiceEditController.php]: Resolved in 82b2a34

File: database/migrations/2018_06_27_201645_v2_0_0_alpha2.php (hunk 10 above)
Comment (UnderDogg) [reply database/migrations/2018_06_27_201645_v2_0_0_alpha2.php]: Resolved in 3aeeb66

File: app/Modules/Layouts/Views/layouts/_header.blade.php (hunk 9 above)
Comment (UnderDogg) [reply app/Modules/Layouts/Views/layouts/_header.blade.php]: Actually ... the addons.index is working at the moment, so re-committed this file with another commit message. The custom_fields route doesn't work, so i deleted that from the menu: 675265b

File: app/Modules/Invoices/Controllers/InvoiceCopyController.php (hunk 5 above)
Comment (UnderDogg) [reply app/Modules/Invoices/Controllers/InvoiceCopyController.php]: This is a very interesting one. Copying invoices does not work properly. When you do not have custom fields the copying does not work at all. Right now it does work, but still needs to be looked at. I've committed the removing of custom fields (was commented out) in c0e0f9f but I think we should look at this as a separate bug report

**Comment by UnderDogg** [comment]
I've made all the commits, 1 commit per remark. Most important is the copying of invoices. It still doesn't work right and needs to be looked at separately. See my commit message in commit c0e0f9f

File: app/Modules/Layouts/Views/layouts/_header.blade.php [hunk 11]
```diff
@@ -28,9 +28,8 @@
                         <i class="fa fa-cog"></i>
                     </a>
                     <ul class="dropdown-menu">
-                        {{--<li><a href="{{ route('addons.index') }}">@lang('ip.addons')</a></li>--}}
+                        <li><a href="{{ route('addons.index') }}">@lang('ip.addons')</a></li>
                         <li><a href="{{ route('currencies.index') }}">@lang('ip.currencies')</a></li>
-                        {{--<li><a href="{{ route('customFields.index') }}">@lang('ip.custom_fields')</a></li>--}}
```
Comment (Kovah) [inline app/Modules/Layouts/Views/layouts/_header.blade.php]: Custom fields are gone now?! :(

**Pull request merged**